src/model/Building.cpp

```diff
@@ -932,6 +933,30 @@ namespace model {
       return Transformation::rotation(Vector3d(0, 0, 1), -degToRad(this->northAxis()));
     }
 
+    double Building_Impl::exteriorPerimeter() {
+
+      Point3dVectorVector polygons;
+
+      for (const auto& space : model().getConcreteModelObjects<Space>()) {
+        Transformation spaceTransformation = space.transformation();
+        for (const auto& surface : space.surfaces()) {
+          if (surface.outsideBoundaryCondition() == "Ground" || surface.surfaceType() == "Floor") {
+            Point3dVector points = spaceTransformation * surface.vertices();
+            if (!points.empty() && points[0].z() == 0.0) {
```

Comment:
Might want to check that all the points have z == 0?

Reply:
Another approach would be to project every floor onto z=0 by just setting z = 0 on all the polygons